pandas/tests/indexes/test_multi.py

```diff
@@ -3307,3 +3307,39 @@ def test_duplicate_multiindex_labels(self):
         with pytest.raises(ValueError):
             ind.set_levels([['A', 'B', 'A', 'A', 'B'], [2, 1, 3, -2, 5]],
                            inplace=True)
+
+    def test_use_enum_in_multiindex(self):
+        # GH 21298
+        # Allow use of Enums as one of the factors in a MultiIndex.
+        from enum import Enum
+        from pandas.core.algorithms import take_1d
+
+        # See https://github.com/pandas-dev/pandas/blob/
+        # 0c65c57a279e755ab7093db925d1e580f9878dae/pandas/util/testing.py#L793-L799.
+        # We cannot simply use tm.assert_index_equal, as the "expected" index
+        # cannot actually be built yet.
+        def _get_ilevel_values(index, level):
+            # accept level number only
+            unique = index.levels[level]
+            labels = index.labels[level]
+            filled = take_1d(
+                unique.values,
+                labels, fill_value=unique._na_value
+            )
+            values = unique._shallow_copy(filled, name=index.names[level])
+            return values
```

Comment:
Given that you use this function only once, I think this abstraction is unnecessary as @jreback alluded to.

Reply:
Ok, amended. Thanks for your guidance.